### PR TITLE
GPII-2147: Replacing "localhost" with 127.0.0.1

### PR DESCRIPTION
**Pull request by kaspermarkus**

### CODE REVIEW

File: lib/dataSource-url.js [hunk 1]
```diff
@@ -94,6 +94,9 @@ kettle.dataSource.URL.requestOptions = ["url", "protocol", "host", "hostname", "
 kettle.dataSource.URL.prepareRequestOptions = function (componentOptions, cookieJar, userOptions, permittedOptions, directModel, userStaticOptions, resolveUrl) {
     var staticOptions = fluid.filterKeys(componentOptions, permittedOptions);
     var requestOptions = fluid.extend(true, {headers: {}}, userStaticOptions, staticOptions, userOptions);
+    if (requestOptions.hostname === "localhost" || requestOptions.host === "localhost") {
```
Comment (amb26) [inline lib/dataSource-url.js]: Very good - please leave a comment reference to GPII-2147 so future readers can understand the issue

Comment (amb26) [inline lib/dataSource-url.js]: Also a tiny test would be helpful - it doesn't need to be particularly realistic, but we should at least have something which verifies that this code path hasn't rotted (and will show up in a coverage check)

**Comment by kaspermarkus** [comment]
let me know if the test is too unrealistic!

**Comment by amb26** [comment]
Hi, your test is fine - unfortunately this pull introduces a test failure
```
13:52:22.769:  jq: FAIL: Module "Kettle JSON parsing Tests" Test name "KETTLE-34 request option resolution test" - Message: Resolved expected requestOptions
13:52:22.769:  jq: Expected: {
    "protocol": "http:",
    "port": 998,
    "path": "/device",
    "host": "localhost"
}
13:52:22.771:  jq: Actual: {
    "port": 998,
    "protocol": "http:",
    "host": "127.0.0.1",
    "path": "/device"
}```

**Comment by kaspermarkus** [comment]
@amb26 sorry about that antranig - tests and linting is passing properly now

**Pull request merged**